shared/js/background/message-handlers.js

```diff
@@ -87,12 +87,21 @@ export function getTopBlockedByPages (options) {
 }
 
 // Click to load interactions
-export async function initClickToLoad (config, sender) {
+export async function initClickToLoad (incomingConfig, sender) {
     await settings.ready()
     const tab = tabManager.get({ tabId: sender.tab.id })
 
     // Remove first-party entries.
     await startup.ready()
+
+    // Only return config for ClickToLoadConfig configured in storage
```

Comment:
Could you explain the idea here? How could the configuration that this function receives be different from the configuration that's in storage?

Also, I find this implementation kind of obtuse. Am I right in saying we are creating a new Object, that has only the keys from the original Object that are also found in tdsStorage? If so, could we not just use a for loop instead?

```javascript
const config = {};
for (let entity in incomingConfig) {
  if (tdsStorage.ClickToLoadConfig[entity]) {
    config[entity] = incomingConfig[entity];
  }
}
```

Maybe I'm misunderstanding though?

Reply:
This configuration is now different because part of the CTP configuration is now hard-coded in [click-to-play.js](https://github.com/duckduckgo/content-scope-scripts/blob/main/src/features/click-to-play.js#L335) in C-S-S, meanwhile the surrogates and blocking rules still live in `social_ctp_configuration.json`. `tdsStorage.ClickToLoadConfig` data comes from `social_ctp_configuration.json` but the selectors and logic to find embedded content in the page and replace them lives in [click-to-play.js](https://github.com/duckduckgo/content-scope-scripts/blob/main/src/features/click-to-play.js#L335), so they can be mismatched. It happened with me when I started the migration, I was replacing the videos with the placeholders, but the videos were not being blocked until I updated `social_ctp_configuration.json`. Yes, it is very confusing and this makes for an obtuse solution from my side :/

But I'm not quite sure how to handle then. I could just comment out the [yt config](https://github.com/duckduckgo/content-scope-scripts/pull/196/files#diff-658f5706013e4033baa9dd3e8885fcbd34cf73ea331a3bfd34764a39b0d0186aR950) in `click-to-play.js` and it would me this not necessary anymore, but we'll have to sync both config releases.

Regarding the for loop suggestion, I agree, it's easier to read it.

Reply:
Ah right, I see. Yes, this is a mess currently, I'm sorry you're being stuck in the middle of it :(.

I think your implementation here is pretty solid now, let's just improve the names and comments to make the purpose clearer:
- How about we rename `incomingConfig` to `contentScopeConfig`?
- Maybe change the comment to say something like "The Click to Load configuration is in the process of being migrated, but is currently split between tdsStorage.ClickToLoadConfig (aka social_ctp_configuration.json), the block list (aka tds.json), and contentent-scope-scripts/features/click-to-play.js. Take care to remove any entries from the content-scope-scripts configuration that aren't also found in tdsStorage.ClickToLoadConfig - otherwise the feature can end up in an inconsistent state."